### PR TITLE
Format job requirements as hours and minutes

### PR DESCRIPTION
**Pull request by irismessage**
## About the PR

Role requirements now show the playtime needed in the format hhH mmM, the same as the playtime stats page.

## Why / Balance

This makes them much more readable. Before I was always having to mentally convert them to hours.

## Technical details

I copied the `ConvertTimeSpanToHoursMinutes` method from `PlayTimeStatsWindow.cs`. I wasn't sure if there's a utils place of some kind to move it to in Shared. In any case it's maybe four duplicated lines of code

## Media

Screenshot_20240707_125630

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

None

**Changelog**

:cl:
- tweak: Made role requirements display in a more understandable format.

### CODE REVIEW

File: Content.Shared/Roles/JobRequirements.cs [hunk 1]
```diff
@@ -224,5 +235,14 @@ public static class JobRequirements
...
         }
+
+        private static string ConvertTimeSpanToHoursMinutes(TimeSpan timeSpan)
+        {
+            var hours = (int) timeSpan.TotalHours;
+            var minutes = timeSpan.Minutes;
+
+            var formattedTimeLoc = Loc.GetString("role-timer-time-format", ("hours", hours), ("minutes", minutes));
+            return formattedTimeLoc;
+        }
```
Comment (metalgearsloth) [inline Content.Shared/Roles/JobRequirements.cs:246]: Pretty sure C# has string formatters for this.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Shared/Roles/JobRequirements.cs (hunk 1 above)
Comment (joelsgp) [reply Content.Shared/Roles/JobRequirements.cs:246]: That part's copied from `PlayTimeStatsWindow.cs`,  I guess whoever wrote it wants the H and M to be localizable.

Comment (Tayrtahn) [reply Content.Shared/Roles/JobRequirements.cs:246]: Both probably should be converted to something like
```c#
return timeSpan.ToString(Loc.GetString("role-timer-time-format"));
```
Then the Loc string can use anything from [the docs](https://learn.microsoft.com/en-us/dotnet/api/system.timespan.tostring?view=net-8.0#system-timespan-tostring(system-string)).

Comment (joelsgp) [reply Content.Shared/Roles/JobRequirements.cs]: [4920b73](https://github.com/space-wizards/space-station-14/pull/29796/commits/4920b730ce06249463d26814dce5259899b44bbf)

**Comment by joelsgp** [comment]
Can I get a review on this @metalgearsloth

**Comment by Plykiya** [comment]
> Can I get a review on this @metalgearsloth

under the list of reviewers you can click the button next to sloth's name to officially request the re-review

**Comment by joelsgp** [comment]
I have "Allow edits by maintainers" enabled, for me it says "These conflicts are too complex to resolve in the web editor". Should I rebase or something

**Comment by metalgearsloth** [comment]
ed moved the switch case to classes. I'll try pushing again later worst case I just open a new branch and merge it.

**Comment by metalgearsloth** [comment]
I ended up replacing it https://github.com/space-wizards/space-station-14/pull/29796

**Pull request closed without merging**